pilot/pkg/networking/core/v1alpha3/gateway.go

```diff
@@ -93,36 +94,31 @@ func (ml *MutableGatewayListener) build(builder *ListenerBuilder, opts gatewayLi
 			if len(opt.networkFilters) > 0 {
 				// this is the terminating filter
 				lastNetworkFilter := opt.networkFilters[len(opt.networkFilters)-1]
-
-				for n := 0; n < len(opt.networkFilters)-1; n++ {
-					ml.Listener.FilterChains[i].Filters = append(ml.Listener.FilterChains[i].Filters, opt.networkFilters[n])
-				}
-				ml.Listener.FilterChains[i].Filters = append(ml.Listener.FilterChains[i].Filters, chain.TCP...)
-				ml.Listener.FilterChains[i].Filters = append(ml.Listener.FilterChains[i].Filters, lastNetworkFilter)
+				filterChain.Filters = append(filterChain.Filters, opt.networkFilters[:len(opt.networkFilters)-1]...)
+				filterChain.Filters = append(filterChain.Filters, opt.tcpAuthFilters...)
```

Comment:
Looks this can be further improved, to make Filters append before. So we can make it simpler to control the filter orders.